logic/overworld.py

```diff
@@ -231,6 +233,7 @@ def __init__(self, options, world_setup, r):
 
         Location().add(HeartPiece(0x078)).connect(bay_water, FLIPPERS)  # in the moat of the castle
         castle_inside = Location()
+        castle_inside.connect(castle_outside, None, one_way=True) # the button in the connector allows access to the castle grounds in ER
```

Comment:
I don't think this will work correctly for ER. As this now says, if you can access the inside of the castle, you can access the outside as well. But, if ER puts the castle in mabe, and you don't have the power bracelet yet. The logic will say you can access outside castle, which isn't true.

Reply:
It's sort of the same problem as the raft house and the angler keyhole. So you need some kind of "dummy" item and location to say "if you can access the casltle, you get the `CASTLE_BUTTON`" and "outside castle can be accessed from ukuku_prairie if you have the `CASTLE_BUTTON`"

Reply:
Oh, you're right. I didn't think that through - focused too much on making the no-requirement one ways work and not enough on the actual logic, haha.

Reply:
Ok, I added a CASTLE_BUTTON item to address this. I also set the house warp up to send you inside the starting house if you start at a one way entrance. D7 is missing because you get stuck after exiting.

Reply:
(also just tacked on a D8 superjump that Jill pointed out)